### PR TITLE
feat: add chat button on hackathon page  

### LINKED ISSUE #7138: Add chat icon to sponsor area in hackathon view
### User Story

As a hacker I'd like an easy way to chat with sponsors

### Why Is this Needed

If we make it easy to chat with sponsors, hackers will get their questions answered easily resulting in  high quality projects

### Description

[comment]: # (Feature or Bug? i.e Type: Bug)
*Type*: Feature

### Current Behavior
no chat button

### Expected Behavior
add chat button

### Definition of Done

- [ ] chat button added to hackathon view right rail next to follow
- [ ] add chat button on org townsquare left rail in the follow section

### Data Requirements
[comment]: # (How will we measure the success of this feature? What kind of tracking is needed for this feature (clicks, impressions, flag)?)

### Additional Information
[comment]: # (Any other information that would be useful, content, screenshots, etc.)

### PR DESCRIPTION
**Pull request by developerfred**
- [x] chat button added to hackathon view right rail next to follow
- [x] add chat button on org townsquare left rail in the follow section

##### Description
Screen Shot 2020-07-28 at 14 11 02

##### Refers/Fixes
close #7138

##### Testing
screen test -> https://share.getcloudapp.com/Qwu0qdWw

### CODE REVIEW

File: app/assets/v2/css/buttons.css [hunk 1]
```diff
@@ -1,13 +1,13 @@
 /*
-	https://developer.mozilla.org/en/docs/Web/CSS/box-shadow
```
Comment (danlipert) [inline app/assets/v2/css/buttons.css:2]: @developerfred can you not include any auto-formatting changes from your IDE? thanks!

Comment (developerfred) [reply app/assets/v2/css/buttons.css:2]: sure!

**Comment by danlipert** [comment]
@androolloyd Does this conflict with your chat sidebar changes?

**Comment by thelostone-mc** [comment]
cc @androolloyd  + @PixelantDesign

**Comment by PixelantDesign** [comment]
@developerfred thanks for this PR. I think we only need the chat button on the right hand side sidebar.

**Comment by danlipert** [comment]
@androolloyd since you are the chat czar, can you give this a check and let us know what you think?

File: app/assets/v2/js/open-chat.js [hunk 2]
```diff
@@ -4,9 +4,26 @@ const openChat = () => {
...
+      const url = user ? `https://chat.gitcoin.co/gitcoin/messages/@${user}` :
+        'https://chat.gitcoin.co/';
 
-      chatWindow = window.open(url, 'Loading', 'top=0,left=0,width=400,height=600,status=no,toolbar=no,location=no,menubar=no,titlebar=no');
+      chatWindow = window.open(
+        url, 'Loading',
+        'top=0,left=0,width=400,height=600,status=no,toolbar=no,location=no,menubar=no,titlebar=no');
+    });
+  });
+  $('[data-sponser-openchat').each(function(index, elem) {
```
Comment (androolloyd) [inline app/assets/v2/js/open-chat.js:15]: how this connects to chat is changed, there is is a new global variable available at window.chatSidebar and a helper method to open up in the new chat window. 

window.chatSidebar.chatWindow(channel)

remove the host from the url and pass just the channel like so into the helper
```window.chatSidebar.chatWindow(`gitcoin/messages/channels/company-${sponserChannel}``)```

**Comment by PixelantDesign** [comment]
@connoroday it should open in the flyout. @androolloyd will this automatically happen given  that's the current implementation? Or is there something @developerfred will need to adjust?

**Review by PixelantDesign — APPROVED** [review]

**Review by frankchen07 — APPROVED** [review]
LGTM.

Just a side note on the original ticket. We should make sure we can measure what adding this button to do, and in the larger picture, if it would actually increase the ability to communicate with sponsors. From our interviews, we already know the chat application wasn't working out the way that we wanted it to, so my thinking is linking to it prominently may not be the best solution forward.

That being said, we have a chat button now on the hackathon page, in the future if the chat density shifts to something else, we have the ability to move the linking of the chat button to the right place. Basically, I think the problem exists at a deeper level, and the button may help, but it could be that the solution to "connect with sponsors" is not chat, just something else that we haven't thought of.

**Comment by thelostone-mc** [comment]
Is this something which we still need to get in  @frankchen07 / @PixelantDesign 
cc @androolloyd

**Comment by PixelantDesign** [comment]
yes please!

**Comment by PixelantDesign** [comment]
@developerfred could we remove the chat button on the left rail?

**Comment by developerfred** [comment]
done @PixelantDesign

**Comment by thelostone-mc** [comment]
Closing this out as chat has been retired

**Pull request closed without merging**